auth/auth/driver/driver.py

```diff
@@ -114,6 +114,7 @@ async def create(self, name, namespace, data):
                     for k, v in data.items()
                 }))
         self.name = name
+        self.namespace = namespace
```

Comment:
This interface seems confusing to me. Why would we initialize the name and namespace in the class if we're just going to use the one the user passes in `create`?

Reply:
Cotton wrote this, some I'm inferring his intentions, but:

K8sSecretResource can represent a Secret that exists or does not yet exist. The top of this method asserts both name and namespace are `None`. When we're creating a new secret we do this:
```python
        tokens_secret = K8sSecretResource(k8s_client)
        cleanup.append(tokens_secret.delete)
        await tokens_secret.create(tokens_secret_name, BATCH_PODS_NAMESPACE, {
```
So even if `create` works well enough to create a k8s resource, but somehow triggers an error afterwards, we'll still delete the secret.

If I was designing this, I'd probably create a factory method:
```python3
@staticmethod
def create(k8s_client, name, namespace, data):
    try:
        secret = K8sSecretResource(k8s_client, name, namespace)
        await secret._delete()
        await self.k8s_client.create_namespace_secret(...)
        return secret
    except:
        await secret.delete()
```

This makes `create` idempotent and ... failure-proof? Idempotence is about running `f` twice being equivalent to running `f` once. We want one more property: `f` failing ("running" zero times) is equivalent to `f` not being invoked. Moreover *every* instance of `K8sSecretResource` has a well defined name and namespace.

I can transform all the resources in this file to follow that style if you think it is an improvement. You and I (and, over time, Arcturus and Dania) own `auth`, so its future is up to us.

Reply:
Save it for a future PR.